src/libs/Localize/index.ts

```diff
@@ -29,66 +26,60 @@ LocaleListener.connect();
 
 // Note: This has to be initialized inside a function and not at the top level of the file, because Intl is polyfilled,
 // and if React Native executes this code upon import, then the polyfill will not be available yet and it will barf
-let CONJUNCTION_LIST_FORMATS_FOR_LOCALES;
+let CONJUNCTION_LIST_FORMATS_FOR_LOCALES: Record<string, Intl.ListFormat>;
 function init() {
-    CONJUNCTION_LIST_FORMATS_FOR_LOCALES = _.reduce(
-        CONST.LOCALES,
-        (memo, locale) => {
-            // This is not a supported locale, so we'll use ES_ES instead
-            if (locale === CONST.LOCALES.ES_ES_ONFIDO) {
-                // eslint-disable-next-line no-param-reassign
-                memo[locale] = new Intl.ListFormat(CONST.LOCALES.ES_ES, {style: 'long', type: 'conjunction'});
-                return memo;
-            }
-
+    CONJUNCTION_LIST_FORMATS_FOR_LOCALES = Object.values(CONST.LOCALES).reduce((memo: Record<string, Intl.ListFormat>, locale) => {
+        // This is not a supported locale, so we'll use ES_ES instead
+        if (locale === CONST.LOCALES.ES_ES_ONFIDO) {
             // eslint-disable-next-line no-param-reassign
-            memo[locale] = new Intl.ListFormat(locale, {style: 'long', type: 'conjunction'});
+            memo[locale] = new Intl.ListFormat(CONST.LOCALES.ES_ES, {style: 'long', type: 'conjunction'});
             return memo;
-        },
-        {},
-    );
+        }
+
+        // eslint-disable-next-line no-param-reassign
+        memo[locale] = new Intl.ListFormat(locale, {style: 'long', type: 'conjunction'});
+        return memo;
+    }, {});
 }
 
+type PhraseParameters<T> = T extends (...args: infer A) => string ? A : never[];
+type Phrase<TKey extends TranslationPaths> = TranslationFlatObject[TKey] extends (...args: infer A) => unknown ? (...args: A) => string : string;
+
 /**
  * Return translated string for given locale and phrase
  *
- * @param {String} [desiredLanguage] eg 'en', 'es-ES'
- * @param {String} phraseKey
- * @param {Object} [phraseParameters] Parameters to supply if the phrase is a template literal.
- * @returns {String}
+ * @param [desiredLanguage] eg 'en', 'es-ES'
+ * @param [phraseParameters] Parameters to supply if the phrase is a template literal.
  */
-function translate(desiredLanguage = CONST.LOCALES.DEFAULT, phraseKey, phraseParameters = {}) {
-    const languageAbbreviation = desiredLanguage.substring(0, 2);
-    let translatedPhrase;
-
+function translate<TKey extends TranslationPaths>(desiredLanguage: 'en' | 'es' | 'es-ES' | 'es_ES', phraseKey: TKey, ...phraseParameters: PhraseParameters<Phrase<TKey>>): string {
```

Comment:
The translate function used to take 3 arguments and the third one being an object containing the parameters to be used by the translation phrase. Now the third argument is using the rest syntax and it's an array. Was that change intended?

Also previously the 3rd argument had a default value and now it does not which caused https://github.com/Expensify/App/issues/34668 and https://github.com/Expensify/App/issues/30949

Reply:
Do you know if we can have typescript help us with that? i.e. yield an error if we are missing to pass parameters to a translation function

Reply:
That's already implemented @s77rt, but only in .ts/.tsx files